packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -106,6 +108,14 @@ library LibVerifying {
             unchecked {
                 state.lastVerifiedBlockId += processed;
             }
+
+            if (config.relaySignalRoot) {
+                // Send the L2's signal root to the signal service so other TaikoL1
+                // deployments, if they share the same signal service, can relay the
+                // signal to their corresponding TaikoL2 contract.
+                ISignalService(resolver.resolve("signal_service", false))
+                    .sendSignal(signalRoot);
```

Comment:
This will always store to the same slot I think no? Just checking because I think that is also how it should work, but you wrote "because we always write to new signal slots" so just want to make sure I got this right.

Reply:
The signal service is designed to never rewrite any slots. For each new value of `signalRoot`, it always ends up with a different slot and the actual data stored in the slot is `1`.

Reply:
Ah right I see now, forgot how it worked.